lib/util/state.js

```diff
@@ -53,62 +56,133 @@ export function getActiveError (otpState) {
+  otpState => otpState.filter,
+  getActiveSearchRealtimeResponse,
+  otpState => otpState.useRealtime,
+  (
+    config,
+    nonRealtimeResponse,
+    otpStateFilter,
+    realtimeResponse,
+    useRealtime
+  ) => {
+    // set response to use depending on useRealtime
+    const response = (!nonRealtimeResponse && !realtimeResponse)
+      ? null
+      : useRealtime
+        ? realtimeResponse
+        : nonRealtimeResponse
+    const itineraries = []
+    // keep track of itinerary hashes in order to not include duplicate
+    // itineraries. Duplicate itineraries can occur in batch routing where a walk
+    // to transit trip can sometimes still be the most optimal trip even when
+    // additional modes such as bike rental were also requested
+    const seenItineraryHashes = {}
+    if (response) {
+      response.forEach(res => {
+        if (res && res.plan) {
+          res.plan.itineraries.forEach(itinerary => {
+            // hashing takes a while on itineraries
+            const itineraryHash = hashItinerary(itinerary)
+            if (!seenItineraryHashes[itineraryHash]) {
+              itineraries.push(itinerary)
+              seenItineraryHashes[itineraryHash] = true
+            }
           })
-          return allActiveModes
-        case 'TRANSIT':
-          return itineraryHasTransit(itinerary)
-        case 'CAR':
-          let hasCar = false
-          itinerary.legs.forEach(leg => {
-            if (isCar(leg.mode)) hasCar = true
-          })
-          return hasCar
-        default:
-          console.warn(`Filter (${filter}) not supported`)
-          return true
-      }
-    })
-  // If no sort type is provided (e.g., because batch routing is not enabled),
-  // do not sort itineraries (default sort from API response is used).
-  return !type
-    ? filteredItineraries
-    : filteredItineraries.sort((a, b) => sortItineraries(type, direction, a, b, config))
-}
+        }
+      })
+    }
+    const {filter, sort} = otpStateFilter
+    const {direction, type} = sort
+    const filteredItineraries = itineraries
+      .filter(itinerary => {
+        switch (filter) {
+          case 'ALL':
+            return true
+          case 'ACTIVE':
+            // ACTIVE filter checks that the only modes contained in the itinerary
+            // are 'active' (i.e., walk or bike). FIXME: add micromobility?
+            let allActiveModes = true
+            itinerary.legs.forEach(leg => {
+              if (!isWalk(leg.mode) && !hasBike(leg.mode)) allActiveModes = false
+            })
+            return allActiveModes
+          case 'TRANSIT':
+            return itineraryHasTransit(itinerary)
+          case 'CAR':
+            let hasCar = false
+            itinerary.legs.forEach(leg => {
+              if (isCar(leg.mode)) hasCar = true
+            })
+            return hasCar
+          default:
+            // return true for unrecognized or falsy filter
+            return true
+        }
+      })
```

Comment:
I think this filter stuff is dead code that we can remove. I think we just failed to remove it when the dropdown was pulled out.